### PR TITLE
Desktop 1774 redux defs

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Fixes up a bunch of types, but I got tired and afraid of making some error proned mistake, so a lot of connects are left as todo's with flow issues.

### CODE REVIEW

File: desktop/renderer/remote-pinentry.js [hunk 1]
```diff
@@ -49,13 +50,13 @@ class RemotePinentry extends Component<void, Props, void> {
 }
 
 export default connect(
-  state => ({
+  (state: TypedState, ownProps: {}) => ({
     pinentryStates: state.pinentry.pinentryStates || {},
   }),
-  dispatch => bindActionCreators({
```
Comment (MarcoPolo) [inline desktop/renderer/remote-pinentry.js]: bindActionCreators almost works, but the types it outputs aren't quite right

File: shared/constants/tracker.js [hunk 2]
```diff
@@ -131,6 +131,18 @@ export type TrackerState = {
...
 }
 
+export type NonUserState = {
+  type: 'nonUser',
+  closed: boolean,
+  hidden: boolean,
+  name: string,
+  reason: string,
+  isPrivate: boolean,
+  inviteLink: ?string
```
Comment (chromakode) [inline shared/constants/tracker.js]: Trailing comma here?

File: shared/flow-typed/react-redux.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import type { Dispatch, Store } from 'redux'
+
+declare module 'react-redux' {
```
Comment (chromakode) [inline shared/flow-typed/react-redux.js]: Attribution required?

File: shared/store/configure-store.js [hunk 4]
```diff
@@ -71,6 +71,7 @@ if (closureStoreCheck) {
   middlewares.push(closureCheck)
 }
 
+// $FlowIssue
```
Comment (chromakode) [inline shared/store/configure-store.js]: What's the flow issue? Could you please expand this comment?

**Pull request merged**